src/frontends/tensorflow/src/helper_transforms/embedding_segments_feature_fusing.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "helper_transforms/embedding_segments_feature_fusing.hpp"
+
+#include <memory>
+#include <vector>
+
+#include "helper_ops/sparse_fill_empty_rows.hpp"
+#include "helper_ops/sparse_segment_ops.hpp"
+#include "helper_ops/unique.hpp"
+#include "ngraph/rt_info.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+using namespace std;
+using namespace ov::pass;
+using namespace ov::opset8;
+
+ov::frontend::tensorflow::pass::EmbeddingSegmentSingleFeatureFusion::EmbeddingSegmentSingleFeatureFusion() {
+    // The transformation looks for pattern (sub-graph) that performs extraction of embedding vectors from the
+    // parameters table for object feature values, and sum up these embedding vectors for every object or compute their
+    // mean value. Such sub-graph is met in the Wide and Deep model in case of the SINGLE categorical feature.
+    auto embedding_table_pattern = ov::pass::pattern::any_input();
+    auto input_values_pattern = ov::pass::pattern::any_input();
+    auto input_indices_pattern = ov::pass::pattern::any_input();
+    auto dense_shape_pattern = ov::pass::pattern::any_input();
+    auto default_value_pattern = ov::pass::pattern::any_input();
+
+    auto greaterequal0 =
```

Comment:
Could you create constants on different line so that pattern is easier to read? Also if possible could you add some description of the pattern in comments? like high level overview?

Reply:
Done